pricegraph/src/graph/shortest_paths.rs

```diff
@@ -200,6 +341,7 @@ where
                 }
             }
         }
+        shortest_path_graph.prepare_next_relaxation_step();
```

Comment:
I don't like that we rely on this side-effect heavy method here quite unrelated to where we actually read the predecessor vector. It would be nice if it could be computed on the fly if needed.

Reply:
I'm OK with side-effecty code as private methods for now since this is the "hot-path" for `pricegraph`. I think if we do introduce nice abstractions we should measure their performance, and we can do that incrementally in future PRs.